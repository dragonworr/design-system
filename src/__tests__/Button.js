import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Button, theme } from '..'

describe('Button', () => {
  test('renders', () => {
    const json = renderer.create(<Button theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('size small sets height and font-size', () => {
    const json = renderer.create(<Button size='small' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('height', '32px')
    expect(json).toHaveStyleRule('font-size', '12px')
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('size medium sets height and font-size', () => {
    const json = renderer.create(<Button size='medium' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('height', '40px')
    expect(json).toHaveStyleRule('font-size', '14px')
  })

  test('size large sets height and font-size', () => {
    const json = renderer.create(<Button size='large' theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('height', '48px')
    expect(json).toHaveStyleRule('font-size', '16px')
  })

  test('fullWidth prop sets width to 100%', () => {
    const json = renderer.create(<Button fullWidth theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('width', '100%')
  })

  test('disabled prop sets', () => {
    const json = renderer.create(<Button disabled theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
  })
})
