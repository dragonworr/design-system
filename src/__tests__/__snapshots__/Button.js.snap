// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled prop sets 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

<button
  className="c0"
  disabled={true}
/>
`;

exports[`Button fullWidth prop sets width to 100% 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  width: 100%;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

.c0:hover {
  background-color: #049;
}

<button
  className="c0"
/>
`;

exports[`Button renders 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

.c0:hover {
  background-color: #049;
}

<button
  className="c0"
/>
`;

exports[`Button size large sets height and font-size 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 48px;
  font-size: 16px;
  padding: 0 22px;
}

.c0:disabled {
  opacity: 0.25;
}

.c0:hover {
  background-color: #049;
}

<button
  className="c0"
  size="large"
/>
`;

exports[`Button size medium sets height and font-size 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

.c0:hover {
  background-color: #049;
}

<button
  className="c0"
  size="medium"
/>
`;

exports[`Button size small sets height and font-size 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 32px;
  font-size: 12px;
  padding: 0 12px;
}

.c0:disabled {
  opacity: 0.25;
}

.c0:hover {
  background-color: #049;
}

<button
  className="c0"
  size="small"
/>
`;

exports[`Button without disabled prop sets 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c0:disabled {
  opacity: 0.25;
}

.c0:hover {
  background-color: #049;
}

<button
  className="c0"
/>
`;
